inference-engine/src/gna_plugin/gna_graph_compiler.cpp

```diff
@@ -522,23 +522,12 @@ void GNAGraphCompiler::PowerPrimitive(InferenceEngine::CNNLayerPtr layer) {
     auto input = layer->insData[0].lock();
 
     auto outputs = *layer->outData.begin();
-    uint32_t num_rows_in = InferenceEngine::details::product(begin(input->getDims()), end(input->getDims()));
-    uint32_t num_columns_in = 1;
+    auto reshaped_dims = Get2DReshapedData(input, 8)->getDims();
+    uint32_t num_rows_in = reshaped_dims[1];
+    uint32_t num_columns_in = reshaped_dims[0];
     uint32_t num_rows_out = num_rows_in;
     uint32_t num_padding = ALIGN(num_rows_in, 8) - num_rows_in;
 
-    if (input->getDims().size() > 2 || input->getDims()[0] >= 8) {
```

Comment:
@admitriev-gna added it for power broadcast case. Why is this removed?

Reply:
This code is moved to Get2DReshapedData() function which must be general for Power and Diagonal primitives.